includes/Core/Authentication/Connected_Proxy_URL.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Authentication\Connected_Proxy_URL
+ *
+ * @package   Google\Site_Kit\Core\Authentication
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Authentication;
+
+use Google\Site_Kit\Core\Storage\Setting;
+
+/**
+ * Connected_Proxy_URL class.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Connected_Proxy_URL extends Setting {
+
+	/**
+	 * The option_name for this setting.
+	 */
+	const OPTION = 'googlesitekit_connected_proxy_url';
+
+	/**
+	 * Matches provided URL with the current proxy URL in the settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url URL to match against the current one in the settings.
+	 * @return bool TRUE if URL matches the current one, otherwise FALSE.
+	 */
+	public function matches_url( $url ) {
+		$sanitize   = $this->get_sanitize_callback();
+		$normalized = $sanitize( $url );
+		return $normalized === $this->get();
+	}
+
+	/**
+	 * Gets the callback for sanitizing the setting's value before saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable A sanitizing function.
+	 */
+	protected function get_sanitize_callback() {
+		return function( $url ) {
+			$url = filter_var( $url, FILTER_SANITIZE_URL );
+			$url = wp_parse_url( $url );
+			if ( ! $url ) {
+				return null;
+			}
+
+			$normalized  = ! empty( $url['host'] ) ? $url['host'] : '';
+			$normalized .= ! empty( $url['path'] ) ? $url['path'] : '/';
+
+			return $normalized;
```

Comment:
I had missed this before, but this is actually a bit too vague right now: For storing the current `home_url()`, we should really store the entire value - it needs to be compared as a whole. The only thing that we should do here is "sanitize" it to always include a trailing slash.

The most important part currently missing here is lack of the protocol, which should trigger a URL change / disconnect just like a change in host and path. But even other parts (e.g. port change) should, so I think we can replace this entire closure with returning `trailingslashit`.

Reply:
Got it, replaced with just `trailingslashit`